src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1339,6 +1333,21 @@ void install_signal_handlers() {
     (*end_signal_setting)();
   }
 
+  if (!ReduceSignalUsage) {
+    // This is just for early initialization phase. Intercepting the signal here
+    // reduces the risk that an attach client accidentally forces HotSpot to quit
+    // prematurely. Late initialization will overwrite BREAK_SIGNAL's handler to
+    // UserHandler.
+    // Note that HotSpot does NOT support signal chaining for BREAK_SIGNAL.
+    // We have to set it outside the window bounded by libjsig's
+    // JVM_begin_signal_setting and JVM_end_signal_setting above, because the
+    // window is intended for signals that support chaining. Otherwise libjsig
+    // would prevent us from overwriting BREAK_SIGNAL's handler to UserHandler.
+    // We also use os::signal() and a dummy handler to avoid special-casing
+    // set_signal_handler() and JVM_HANDLE_XXX_SIGNAL().
+    os::signal(BREAK_SIGNAL, CAST_FROM_FN_PTR(void*, DummySIGBREAKHandler));
```

Comment:
Bummer. This is what I tried to avoid. Sorry!

By moving sigaction of BREAK_SIGNAL out of the "window" -- [begin_signal_setting, end_signal_setting], `libjsig` will not record the old action of 'BREAK_SIGNAL', so HotSpot still has freedom to replace it with **UserHandler**, which can handle ThreadDump request.

I agree that your approach is less intrusive and is better than previous one!